CMakeLists.txt

```diff
@@ -118,6 +118,10 @@ if(HAVE_LIBDL)
     target_link_libraries(wiredtiger "dl")
 endif()
 
+if(ENABLE_TCMALLOC AND HAVE_LIBTCMALLOC)
+    target_link_libraries(wiredtiger ${HAVE_LIBTCMALLOC})
```

Comment:
This is probably just my CMake ignorance, but isn't `target_link_libraries` supposed to accept the name of the library that you want the target to link with? But `HAVE_LIBTCMALLOC` seems to be a `ON/OFF` boolean type thing.

Reply:
Good pick up! This would've been true until recently. With the changes of #6720 (WT-7748) the config variables defined by `config_lib` (`HAVE_LIBTCMALLOC` being one of them) are now loaded with the library path, which is needed when we include libraries from non-standard linker paths.

Reply:
We can still use `HAVE_LIBTCMALLOC` as boolean like variable. CMake equates non-empty strings as truthy values and empty strings as falsy values.

Reply:
Thanks! If you have a moment, could you point me at the line where the variable gets loaded with the library path? I looked at the changes to that `config_lib` function but couldn't spot the line where it's doing that.

No need to block the PR for this since it's just for my own learning.

Reply:
Yep, no problem! The config variable gets set here:
https://github.com/wiredtiger/wiredtiger/blob/f6da30bb957f5fee141519070c58d67774ccce6f/build_cmake/helpers.cmake#L452
`${config_name}` is the first parameter from the `config_lib` call. TCMalloc for example:
https://github.com/wiredtiger/wiredtiger/blob/f6da30bb957f5fee141519070c58d67774ccce6f/build_cmake/configs/auto.cmake#L274-L278

---

The actual value is sourced from the `find_library` function, storing the library path into the variable `has_lib_${config_name}`:
https://github.com/wiredtiger/wiredtiger/blob/f6da30bb957f5fee141519070c58d67774ccce6f/build_cmake/helpers.cmake#L426
Which if it exists, is loaded into the `has_lib` variable:
https://github.com/wiredtiger/wiredtiger/blob/f6da30bb957f5fee141519070c58d67774ccce6f/build_cmake/helpers.cmake#L430

Reply:
`has_lib` is probably a point of confusion. It used to just store a boolean value but has since been updated to store the full library path.

Reply:
Winner. Thanks so much.

Yeah, it makes sense now. I think when I saw `has_<foo>`, I thought of it as a boolean when in fact it can store a path and still be used as a boolean due to CMake's advanced type system.

Reply:
One of the things I was thinking when I made that change was that the variables probably ought to be renamed. But I didn't want to propose it right away as it's pretty invasive.

Reply:
Thanks for also checking this out @sauclovian-wt  :)

That's completely fair, renaming the variables right away is probably too invasive. Additionally we also want to maintain the `HAVE_LIB<FOO>` config names since its whats exported by autoconf (as we keep both builds maintained in the short term). Though will definitely create some future work refactoring/renaming things once we deprecated autoconf.